src/commands/deploy.rs

```diff
@@ -398,6 +415,74 @@ impl DeployCommand {
     }
 }
 
+async fn wait_for_ready(
+    address: &str,
+    hippo_url: &str,
+    cfg: &spin_loader::local::config::RawAppManifest,
+    readiness_timeout_secs: u16,
+) {
+    if readiness_timeout_secs == 0 {
+        return;
+    }
+
+    if cfg.components.is_empty() {
+        return;
+    }
+
+    let url_result = Url::parse(hippo_url);
+    let scheme = match &url_result {
+        Ok(url) => url.scheme(),
+        Err(_) => "http",
+    };
```

Comment:
What's the rationale here? It doesn't seem like the hippo URL scheme and app host scheme would have anything to do with each other.

Reply:
This is the logic we use in `print_available_routes` - if it's wrong we need to update it there too.  I'll see if there's a way for Hippo to report the scheme, but I think the idea was that local platforms were typically non-TLS for both Hippo and apps, and remote platforms typically TLS for both.

Reply:
Ah, so its hacks then.

I think it would be less confusing if both this and `print_available_routes` took an e.g. `app_base_url` which could be built separately with the hack documented.

Given that this is copying an existing pattern it doesn't need to block this PR.